src/SQLiteAsync.cs

```diff
@@ -77,7 +79,10 @@ public SQLiteAsyncConnection (string databasePath, bool storeDateTimeAsTicks = t
 		public SQLiteAsyncConnection (string databasePath, SQLiteOpenFlags openFlags, bool storeDateTimeAsTicks = true)
 		{
 			_openFlags = openFlags;
+			isFullMutex = _openFlags.HasFlag (SQLiteOpenFlags.FullMutex);
 			_connectionString = new SQLiteConnectionString (databasePath, storeDateTimeAsTicks);
+			if(isFullMutex)
+				_fullMutexReadConnection = new SQLiteConnectionWithLock (_connectionString, openFlags) { SkipLock = true };
```

Comment:
This opens a new connection and already executes the PRAGMA for WAL. I also added my PRAGMA for sqlcipher in this constructor.

So that means it immediately opens a connection and executes 2 queries whenever you request a connection, even if you're not using this connection itself (this is the case when you only write to the DB). This is why I asked why the full mutex flag has become the default... I request a lot of connections which used to be pooled.